matrix_client/crypto/olm_device.py

```diff
@@ -54,3 +54,41 @@ def sign_json(self, json):
             json['unsigned'] = unsigned
 
         return json
+
+    def verify_json(self, json, user_key, user_id, device_id):
+        """Verifies a signed key object's signature.
+
+        The object must have a 'signatures' key associated with an object of the form
+        `user_id: {key_id: signature}`.
+
+        Args:
+            json (dict): The JSON object to verify.
+            user_key (str): The public ed25519 key which was used to sign the object.
+            user_id (str): The user who owns the device.
+            device_id (str): The device who owns the key.
+
+        Returns:
+            True if the verification was successful, False if not.
+
+        Raises:
+            KeyError: Wrong json format.
+        """
+        signatures = json.pop('signatures')
+
+        key_id = 'ed25519:{}'.format(device_id)
+        signature_base64 = signatures[user_id][key_id]
+        unsigned = json.pop('unsigned', None)
+
+        try:
+            olm.ed25519_verify(user_key.encode(),
+                               encode_canonical_json(json),
+                               signature_base64.encode())
```

Comment:
Let's be explicit in this call site that we're encoding as utf-8 for clarity `.encode(encoding="utf-8")`. Same thing anywhere we're decoding.

Reply:
This one is weird because .encode() and .decode() are not necessary in Python2 since str and bytes are the same. I have to add them for Python3 support, and in Python3, 'utf-8' is the default encoding, so no need to specify it... Is there really a case where something could go wrong?  
Wondering this because calling .encode()/decode() each time we interact with the bindings already feels clumsy enough.

Reply:
Definitely not necessary. Including the `encoding` kwarg doesn't materially change anything, so I'm not dead set on it if you're opposed. That said, I strongly disagree on `encode`/`decode` being clumsy here. There's many ways to represent a string as a sequence of bytes, and I'm glad that python3 forces you to be explicit about that transformation.

Reply:
Sure, the clumsy part was the "each time", meaning that this sort of thing should be handled under the hood. Guess what, yesterday someone came up with a project that addresses this kind of things https://github.com/poljar/python-olm (last discussion in #matrix-dev atm). I'm waiting on Richard's opinion before switching.

Reply:
I definitely like the idea of those kind of conversions happening under the hood in the python olm bindings. I like the idea even more of the python bindings living outside the sdk in a package hosted on pypi. I'd probably be against switching to those bindings unless we get the blessing from @richvdh to just move the python bindings outside the olm library entirely and maintain them as a separate package.